lib/cequel/metal/data_set.rb

```diff
@@ -565,6 +567,22 @@ def consistency(consistency)
         end
       end
 
+      def page_size(page_size)
+        clone.tap do |data_set|
+          data_set.query_page_size = page_size
+        end
+      end
+
+      def paging_state(paging_state)
+        clone.tap do |data_set|
+          data_set.query_paging_state = paging_state
+        end
+      end
+
+      def next_paging_state
+        execute_cql(*cql).paging_state
+      end
```

Comment:
You are correct, the query would be called a second time.  I can take a whack at the memoization.